File: Serene/Serene.Script/Imports/FormContexts/Northwind.OrderDetailForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class OrderDetailForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.OrderDetail";
    
        public OrderDetailForm(string idPrefix) : base(idPrefix) {}
    
        public LookupEditor ProductID { [InlineCode("{this}.w('ProductID', Serenity.LookupEditor)")] get; private set; }
        public DecimalEditor UnitPrice { [InlineCode("{this}.w('UnitPrice', Serenity.DecimalEditor)")] get; private set; }
        public IntegerEditor Quantity { [InlineCode("{this}.w('Quantity', Serenity.IntegerEditor)")] get; private set; }
        public DecimalEditor Discount { [InlineCode("{this}.w('Discount', Serenity.DecimalEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.OrderDetailForm.ts
namespace Serene.Northwind {
    export class OrderDetailForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.OrderDetail';

    }

    export interface OrderDetailForm {
        ProductID: Serenity.LookupEditor;
        UnitPrice: Serenity.DecimalEditor;
        Quantity: Serenity.IntegerEditor;
        Discount: Serenity.DecimalEditor;
    }

    [['ProductID', Serenity.LookupEditor], ['UnitPrice', Serenity.DecimalEditor], ['Quantity', Serenity.IntegerEditor], ['Discount', Serenity.DecimalEditor]].forEach(x => Object.defineProperty(OrderDetailForm.prototype, <string>x[0], { get: function () { return this.w(x[0], x[1]); }, enumerable: true, configurable: true }));
}

